src/main/java/org/variantsync/diffdetective/diff/difftree/parse/DiffTreeParser.java

```diff
@@ -12,21 +12,64 @@
 import org.variantsync.diffdetective.diff.difftree.DiffNode;
 import org.variantsync.diffdetective.diff.difftree.DiffTree;
 import org.variantsync.diffdetective.diff.difftree.DiffType;
+import org.variantsync.diffdetective.diff.difftree.NodeType;
 import org.variantsync.diffdetective.diff.result.DiffError;
 import org.variantsync.diffdetective.diff.result.DiffParseException;
+import org.variantsync.diffdetective.feature.CPPAnnotationParser;
 import org.variantsync.diffdetective.util.Assert;
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.StringReader;
 import java.util.Stack;
+import java.util.regex.Pattern;
 
 /**
  * Parser that parses {@link DiffTree}s from text-based diffs.
+ *
+ * Note: Weird line continuations and comments can cause misidentification of conditional macros.
+ * The following examples are all correct according to the C11 standard: (comment end is marked by
+ * {@code *\/}):
+ * <code>
+ * /*
+ * #ifdef A
+ * *\/
+ *
+ * #ifdef /*
+ * *\/ A
+ * #endif
+ *
+ * # /**\/ ifdef
+ * #endif
+ *
+ * # \
+ *   ifdef
+ * #endif
+ * </code>
  */
 public class DiffTreeParser {
+    /** Matches the beginning of conditional macros.
+     * It doesn't match the whole macro name, for example for {@code #ifdef} only {@code "#if"} is
+     * matched and only {@code "if"} is captured.
+     *
+     * Note that this pattern doesn't handle comments between {@code #} and the macro name.
+     */
+    private final static Pattern macroPattern =
+        Pattern.compile("^[+-]?\\s*#\\s*(if|elif|else|endif)");
+
+    /* Settings */
+    final boolean collapseMultipleCodeLines;
+    final boolean ignoreEmptyLines;
+    final CPPAnnotationParser annotationParser;
+
+    /* State */
+    private final Stack<DiffNode> beforeStack = new Stack<>();
+    private final Stack<DiffNode> afterStack = new Stack<>();
+
+    private DiffNode lastArtifact = null;
```

Comment:
Please document what this field is used for.

Reply:
I added documentation for all fields. (I remember that I wanted to do that anyway, but forgot it somehow.)